Loop/AppDelegate.swift

```diff
@@ -27,62 +27,108 @@ final class AppDelegate: UIResponder, UIApplicationDelegate, DeviceOrientationCo
     private var bluetoothStateManager: BluetoothStateManager!
 
     var window: UIWindow?
+    
+    var launchOptions: [UIApplication.LaunchOptionsKey: Any]?
 
     private var rootViewController: RootNavigationController! {
         return window?.rootViewController as? RootNavigationController
     }
-
-    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+    
+    private var isAfterFirstUnlock: Bool {
```

Comment:
Yay!

[super-nit] it always felt a teeny bit weird to me that this was computed variable with a side-effect of writing a file.  IMHO I'd change it to 
```swift
private func isAfterFirstUnlock() -> Bool {
```